src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3234,70 +3325,147 @@ public static void createA1A6ToStaticNets(Design design) {
-            gndInvertibleToVcc = design.getGndNet();
-        }
         Map<String, Pair<String, String>> pinMapping = belTypeSitePinNameMapping.get(series);
-        final String[] pins = new String[] {"CE", "SR"};
-        for (Cell cell : design.getCells()) {
-            if (isUnisimFlipFlopType(cell.getType())) {
-                SiteInst si = cell.getSiteInst();
+        Net vccNet = design.getVccNet();
+        if (series == Series.Versal) {
+            // In Versal, sitewires for a SLICE's CE pins are not assigned to the VCC net
+            // Assume that the lack of sitewire for a placed FF indicates VCC
+            for (SiteInst si : design.getSiteInsts()) {
                 if (!Utils.isSLICE(si)) {
                     continue;
                 }
-                BEL bel = cell.getBEL();
-                Pair<String, String> sitePinNames = pinMapping.get(bel.getBELType());
-                for (String pin : pins) {
-                    BELPin belPin = cell.getBEL().getPin(pin);
-                    Net net = si.getNetFromSiteWire(belPin.getSiteWireName());
-                    if (net == null || (net == gndInvertibleToVcc && pin.equals("SR"))) {
-                        String sitePinName;
-                        if (pin.equals("CE")) { // CKEN
-                            sitePinName = sitePinNames.getFirst();
-                        } else { //SRST
-                            sitePinName = sitePinNames.getSecond();
+                for (Cell cell : si.getCells()) {
+                    BEL bel = cell.getBEL();
+                    if (bel == null || !bel.isFF()) {
+                        continue;
+                    }
+
+                    if (!bel.getBELType().equals("FF")) {
+                        assert(bel.getBELType().matches("(SLICE_IMI|SLICE[LM]_IMC)_FF(_T)?"));
+                        continue;
+                    }
+
+                    Pair<String, String> sitePinNames = pinMapping.get(bel.getName());
+                    final String[] belPinNames = new String[] {"CE", "SR"};
+                    for (String belPinName : belPinNames) {
+                        String sitePinName = (belPinName == belPinNames[0]) ? sitePinNames.getFirst() : sitePinNames.getSecond();
+                        SitePinInst spi = si.getSitePinInst(sitePinName);
+                        if (spi != null) {
+                            if (belPinName == belPinNames[0]) {
+                                // CE
+                                continue;
+                            }
+                            // SR
+                            if (!spi.getNet().isGNDNet()) {
+                                continue;
+                            }
+                        }
+
+                        Net net = si.getNetFromSiteWire(sitePinName);
+                        if (net != null) {
+                            if (belPinName == belPinNames[0]) {
```

Comment:
`belPinNames[0]` could be replaced with `CE` if we use the refactoring above.  This makes for more readable code.